pkg/reconciler/events/cloudevent/cloud_event_controller.go

```diff
@@ -144,7 +145,9 @@ func SendCloudEventWithRetries(ctx context.Context, object runtime.Object) error
 	_, isRun := object.(*v1alpha1.Run)
 
 	wasIn := make(chan error)
+	e.WaitGroup.Add(1)
```

Comment:
One thing that I'm noticing here is that the interface is really just a wrapper around a waitGroup, i.e. it is pretty similar to the original implementation that just used a plain waitGroup. I'm sorry if I led you down a rabbit hole with my suggestions about how this interface should be implemented, but now that I'm seeing it in a PR we might just want a wrapper around the CloudEventClient. Something like

```
type OurOwnCloudEventClient struct {
   client *cloudEvents.Client
   eventCount int
}

func initialize(ce *cloudEvents.Client) *OurOwnCloudEventClient {
   client = ce
   eventCount = 0
}

func (c *OurOwnCloudEventClient) SendCloudEventWithRetries(ctx, object) {
  <do some stuff>
  eventCount += 1
  go func() {
    c.client.Send()
  }
}

func (c *OurOwnCloudEventClient) checkEventsOrdered(t *testing.T, <some other args>) {
    eventsSent := []string{}
    for i := 0; i < c.eventCount, i ++ {
      e <- c.client.Events
      eventsSent = append(eventsSent, e)
    }
   // Compare expected list of events with actual list of events
}
```
In this example, eventCount can also be replaced with a waitgroup. WDYT?